src/serializer/functions.js

```diff
@@ -147,6 +132,28 @@ export class Functions {
     return recordedAdditionalFunctions;
   }
 
+  __generateInitialOptimizedFunctionsFromRealm(): Array<AdditionalFunctionEntry> {
+    let realm = this.realm;
+    let recordedAdditionalFunctions = [];
+    realm.optimizedFunctions.forEach(entry => {
+      let value = entry.value instanceof AbstractValue ? this._unwrapAbstract(entry.value) : entry.value;
```

Comment:
You no longer have to do the unwrapping at this point. Do it earlier when adding to `optimizedFunctions`, and make it a map of `ECMAScriptSourceFunctionValue` to an optional `ArgModel`.